src/coefficients/stdfunction_coefficient.hpp

```diff
@@ -0,0 +1,165 @@
+
+#include <functional>
+#include "mfem.hpp"
+
+/**
+   \brief StdFunctionCoefficient is an easy way to make an mfem::Coefficient using a lambda
+
+   This is a place holder until the coefficient of the same name is merged into mfem.
+
+*/
+class StdFunctionCoefficient : public mfem::Coefficient
+{
+public:
+  /// Constructor that takes in an mfem Vector representing the coordinates and produces a double
+  StdFunctionCoefficient(std::function<double(mfem::Vector &)> func);
+
+  virtual double Eval (mfem::ElementTransformation &Tr, const mfem::IntegrationPoint &ip);
+private:
+  std::function<double(mfem::Vector &)> m_func;
+};
+
+/**
+   \brief StdFunctionVectorCoefficient is an easy way to make an mfem::Coefficient using a lambda
+
+*/
+class StdFunctionVectorCoefficient : public mfem::VectorCoefficient
+{
+public:
+
+  /**
+     \brief StdFunctionVectorCoefficient is an easy way to make an mfem::Coefficient using a lambda
+
+     \param[in] dim The dimension of the VectorCoefficient
+     \param[in] func Is a function that matches the following prototype void(mfem::Vector &, mfem::Vector &). The first argument of the function is the position, and the second argument is the output of the function.
+  */
+  StdFunctionVectorCoefficient(int dim,
+                               std::function<void(mfem::Vector &, mfem::Vector &)> func);
+
+  virtual void Eval (mfem::Vector &V, mfem::ElementTransformation &T, const mfem::IntegrationPoint &ip);
+private:
+  std::function<void(mfem::Vector &,mfem::Vector &)> m_func;
+};
+
+/**
+   \brief MakeEssList takes in a mesh, a vector coefficient, and produces a list of essential boundary conditions
+
+   \param[in] m Mesh
+   \param[in] c A VectorCoefficient that is porjected on to the mesh. All d.o.f's are examined and those that are the condition (> 0.) are appended toe the vdof list.
+   \param[out] ess_vdof_list The list of vdofs that should be part of the essential boundary conditions
+*/
+void MakeEssList(mfem::Mesh &m, mfem::VectorCoefficient &c, mfem::Array<int> & ess_vdof_list);
+
+/**
+   \brief This method creates an array of size(local_elems), and assigns attributes based on the coefficient c
+
+   This method is useful for creating lists of attributes that correspond to elements in the mesh
+
+   \param[in] m The mesh
+   \param[inout] attr_list Should be an array that will hold the attributes that correspond to each element
+   \param[in] c The coefficient provided that will be evaluated on the mesh
+   \param[in] digitize An optional function that can be called to assign attributes based on the value of c at a given projection point. By default, values of c at a given d.o.f that are > 0. are assigned attribute 2, otherwise attribute 1.
+
+*/
+void MakeAttributeList(mfem::Mesh &m, mfem::Array<int> &attr_list,
+                       mfem::Coefficient &c,
+                       std::function<int(double)> = [](double v)
+{
+  return v > 0. ? 2 : 1;
+});
+
+/**
```

Comment:
I like this doxygen standard. I'll make myself an issue to get the other files in this format.